starfish/io.py

```diff
@@ -29,6 +30,28 @@ def __init__(self):
         self.backend = None
         self.baseurl = None
 
+    @classmethod
+    def from_data(cls, image_stack: ImageStack, aux_dict: Optional[Mapping[str, ImageStack]]=None):
```

Comment:
classmethods can't self-reference because of how they're initialized (something about how Stack hasn't resolved by the time the function is resolved).

Reply:
yeah, type annotations in that scenario should be a string.

`-> "Stack"`

Reply:
Ah, cool. Also, is fixed in 3.7: https://docs.python.org/dev/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations